src/prettyprint.lua

```diff
@@ -0,0 +1,221 @@
+---@param passed number
+---@param total number
+---@return string rendered
+local function createSummaryInfo(header, failed, passed, total)
+	local stringTable = {}
+	if failed > 0 then
+		tablex.push(
+			stringTable,
+			COLOURS.TESTS_FAILED(("%d failed"):format(failed))
+		)
+	end
+
+	if passed > 0 then
+		tablex.push(
+			stringTable,
+			COLOURS.TESTS_PASSED(("%d passed"):format(passed))
+		)
+	end
+
+	tablex.push(stringTable, ("%d total"):format(total))
+	return ("%s %s"):format(
+		COLOURS.BOLD(header),
+		table.concat(stringTable, ", ")
+	)
+end
+
+--- Prints a simple summary about the tests
+---@param results lest.TestSuiteResults
+local function printSummary(results)
+	local passedTests, failedTests, totalTests = 0, 0, 0
+	local passedSuites, failedSuites, totalSuites = 0, 0, #results
+
+	for _, testSuite in ipairs(results) do
+		if testSuite.pass then
+			passedSuites = passedSuites + 1
+		end
+
+		traverseNodes(testSuite, function(node, _)
+			if node.type == NodeType.Test then
+				totalTests = totalTests + 1
+				if node.pass then
+					passedTests = passedTests + 1
+				end
+			end
+
+			return true
+		end)
+	end
+
+	failedSuites = totalSuites - passedSuites
+	failedTests = totalTests - passedTests
+
+	print(
+		createSummaryInfo(
+			TEST_SUITES_HEADER,
+			failedSuites,
+			passedSuites,
+			totalSuites
+		)
+	)
+
+	print(createSummaryInfo(TESTS_HEADER, failedTests, passedTests, totalTests))
+end
+
+--- Prints detailed error reports about the tests that failed.
+---@param results lest.TestSuiteResults[]
+local function printTestErrors(results)
+	---@type table<number, {displayName: string, result: lest.TestResult}>
+	local failedTests = {}
+
+	---@type table<number, string>
+	local parentNames = {}
+	for _, testSuite in ipairs(results) do
+		if not testSuite.pass then
+			traverseNodes(testSuite, function(node)
+				if node.type == NodeType.Describe then
+					if node.pass then
+						-- Skip traversing this branch as it doesn't have anything that failed.
+						return false
+					end
+
+					tablex.push(parentNames, node.name)
+				elseif node.type == NodeType.Test and not node.pass then
+					tablex.push(failedTests, {
+						displayName = table.concat(parentNames, " › ")
+							.. " › "
+							.. node.name,
+						result = node,
+					})
+				end
+
+				return true
+			end, function()
+				tablex.pop(parentNames)
+			end)
+		end
+	end
+
+	for _, failedTest in ipairs(failedTests) do
+		print(
+			COLOURS.ERROR_HEADER(
+				string.format(" %s %s\n", ERROR_SYMBOL, failedTest.displayName)
+			)
+		)
+
+		printIndentedBlock(3, tostring(failedTest.result.error))
+		print()
+	end
+end
+
+--- Replicates the Jest test header output
+---@param testSuite lest.TestSuiteResults
+local function printHeader(testSuite)
+	local fileNameIndex = testSuite.name:find("/[^/]*$")
+	local renderedName = COLOURS.DIMMED(testSuite.name:sub(1, fileNameIndex))
+		.. COLOURS.FILENAME(
+			testSuite.name:sub(fileNameIndex + 1, #testSuite.name)
```

Comment:
I believe `string.sub` will read until the end of the string if the second argument is not passed.